server/szurubooru/func/tag_categories.py

```diff
@@ -117,6 +121,11 @@ def update_category_color(category: model.TagCategory, color: str) -> None:
     category.color = color
 
 
+def update_category_order(category: model.TagCategory, order: int) -> None:
+    assert category
+    category.order = order
```

Comment:
I think we should verify that the given `order` is a positive integer here, and maybe add an upper bound, too. (Unless we really want the full range of ints to be available?)

Would recommend creating an `InvalidTagCategoryOrderError` exception to go with this as well.

Reply:
I see no reason to restrict it to only positive integers, I could definitely see someone adding a negative ordered category instead of updating the order of all other categories. I'll check what happens when you go over the INT_MAX (not sure if that is a problem for the database), but I generally don't think we should restrict things without a good/technological reason.

Reply:
Ok, sounds reasonable